### PR TITLE
13145, 14225, 14288 - Deck Wreck - make Deck "behave properly" in setup(true)/setup(false) flow by letting DrawPile control its listener adding/removing

### PR DESCRIPTION
**Pull request by Cattlesquat**

### CODE REVIEW

File: vassal-app/src/test/java/VASSAL/build/MockModuleTest.java [hunk 1]
```diff
@@ -24,8 +24,8 @@ public void init() throws Exception {
     final DataArchive arch = mock(DataArchive.class);
 
     when(module.getDataArchive()).thenReturn(arch);
-    doNothing().when(module).incorporateSourcesAndListeners();
-    doNothing().when(module).markModuleListeners();
+    //doNothing().when(module).incorporateSourcesAndListeners();
+    //doNothing().when(module).markModuleListeners();
```
Comment (uckelman) [inline vassal-app/src/test/java/VASSAL/build/MockModuleTest.java:28]: You can just delete these two lines. (And the import of `doNothing` a bit further up.)

Comment (Cattlesquat) [reply vassal-app/src/test/java/VASSAL/build/MockModuleTest.java]: done

File: vassal-app/src/main/java/VASSAL/build/GameModule.java [hunk 2]
```diff
@@ -1111,6 +1116,31 @@ public void dumpNewListeners() {
...
+    }
+  }
+
+  public void restoreNewListeners() {
+    if (!isLoadingContinuationSemaphore()) {
+      Collections.copy(keyStrokeSourcesToAdd, keyStrokeSourcesToAddBackup);
+      Collections.copy(keyStrokeListenersToAdd, keyStrokeListenersToAddBackup);
+      Collections.copy(sideChangeListenersToAdd, sideChangeListenersToAddBackup);
+    }
+  }
```
Comment (jonathanrwatts) [inline vassal-app/src/main/java/VASSAL/build/GameModule.java]: Why do backup and restore use the exact same code?

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/GameModule.java]: that code is no longer in the PR

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/GameModule.java]: (but good point otherwise haha)

**Pull request merged**